portfolio/src/main/java/com/google/sps/data/AUX.java

```diff
@@ -35,22 +45,52 @@ private static void GetDevKey(){
         }
     }
 
-    public static Video VideoIdToObject(String videoId) throws IOException, GeneralSecurityException{
+    public static Video VideoIdToObject(String videoId) throws VideoException{
         GetDevKey();
-        YouTube youtubeService = getService();
-        // Define and execute the API request
-        YouTube.Videos.List videoRequest = youtubeService.videos()
+
+        VideoListResponse videoResponse = new VideoListResponse();
+        try{
+            //Get youtube service
+            YouTube youtubeService = getService();
+
+            // Define and execute the API request
+            YouTube.Videos.List videoRequest = youtubeService.videos()
             .list("snippet");
-        VideoListResponse videoResponse = videoRequest.setKey(DEVELOPER_KEY)
+
+            videoResponse = videoRequest.setKey(DEVELOPER_KEY)
             .setId(videoId)
             .execute();
-        com.google.api.services.youtube.model.Video video = videoResponse.getItems().get(0);
-        String videoTitle = video.getSnippet().getTitle();
-        String videoDescription = video.getSnippet().getLocalized().getDescription();
-        String thumbnailUrl = video.getSnippet().getThumbnails().getDefault().getUrl();
-        boolean isPublic = true;
-        Video result = new Video(videoId, videoTitle, videoDescription, thumbnailUrl, isPublic);
-        return result;
+
+            com.google.api.services.youtube.model.Video video = videoResponse.getItems().get(0);
+            String videoTitle = video.getSnippet().getTitle();
+            String videoDescription = video.getSnippet().getLocalized().getDescription();
+            String thumbnailUrl = video.getSnippet().getThumbnails().getDefault().getUrl();
+            String videoCaptions = getVideoCaptions(videoId);
+            boolean isPublic = true;
+            Video result = new Video(videoId, videoTitle, videoDescription, thumbnailUrl, videoCaptions, isPublic);
+            return result;
+        }
+        catch (IOException e) {
+            throw new VideoException("Unable to retrieve video");
+        } 
+        catch (IndexOutOfBoundsException e) {
+            throw new VideoException("No such video found.");
+        } 
+        catch (GeneralSecurityException e) {
+            throw new VideoException("A security exception occurred");
+        }
+        catch(Exception e){
+            if(DEBUG){
+                //Catch any other error that has been potentially overlooked
+                StringWriter sw = new StringWriter();
+                PrintWriter pw = new PrintWriter(sw);
+                e.printStackTrace(pw);
+            }
+
+            throw new VideoException(sw.toString());
```

Comment:
sw is not necessarily initialized to anything here (it is out of scope). Did you mean to pass in a String pertaining to the Exception e?